soc/xtensa/intel_adsp/Kconfig

```diff
@@ -5,6 +5,7 @@
 
 config SOC_FAMILY_INTEL_ADSP
 	select WINSTREAM
+	select ARCH_SUPPORTS_COREDUMP
```

Comment:
Or maybe replace the ifndef CONFIG_XTENSA in fatal.c with this option instead? Seems entirely reasonable at first glance, but you are more familiar than I am with these.